pkgs/tools/security/sequoia/default.nix

```diff
@@ -0,0 +1,82 @@
+{ stdenv, fetchFromGitLab, lib, darwin
+, git, nettle, llvmPackages, cargo, rustc
+, rustPlatform, pkgconfig, glib
+, openssl, sqlite, capnproto
+, pythonSupport ? true, python37Packages ? null
+}:
+
+assert pythonSupport -> python37Packages != null;
+
+rustPlatform.buildRustPackage rec {
+  pname = "sequoia";
+  version = "0.9.0";
+
+  src = fetchFromGitLab {
+    owner = "sequoia-pgp";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "13dzwdzz33dy2lgnznsv8wqnw2501f2ggrkfwpqy5x6d1kgms8rj";
+  };
+
+  nativeBuildInputs = [
+    pkgconfig
+    cargo
+    rustc
+    git
+    llvmPackages.libclang
+    llvmPackages.clang
+  ];
+  checkInputs = []
+    ++ lib.optionals pythonSupport [
+      python37Packages.pytest
+      python37Packages.pytestrunner
+      python37Packages.setuptools
+    ]
+  ;
+  buildInputs = [
+    openssl
+    sqlite
+    nettle
+    capnproto
+  ]
+    ++ lib.optionals pythonSupport [python37Packages.python python37Packages.cffi]
+    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ]
+  ;
+
+  LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
+
+  # Set the Epoch to 1980; otherwise the Python wheel/zip code 
+  # gets very angry SOURCE_DATE_EPOCH=315532800;
```

Comment:
see `ensureNewerSourcesForZipFilesHook`

Reply:
Boy I really needed to know that for a long time. Thanks!